src/main/java/com/leverx/learningmanagementsystem/core/security/filter/CloudTenantInfoFilter.java

```diff
@@ -22,32 +22,58 @@
 @Profile("cloud")
 public class CloudTenantInfoFilter extends OncePerRequestFilter {
 
+    public static final String BEARER_PREFIX = "Bearer ";
+    public static final String ZID = "zid";
+    public static final String AUTHORIZATION = "Authorization";
+    public static final String API_NAME = "-lms";
+
     @Override
     protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
         String path = request.getRequestURI();
-        if (path.startsWith("/actuator")) {
+        if (path.startsWith("/actuator") || path.startsWith("/api/v1")) {
```

Comment:
each filter must be responsible only for one job, here you doing redundant checks and logic that should be done in another filters. To avoid it, you have to define order for this filter which should be after your `apiSecurityFilterChain`. Spring provides FilterRegistrationBean to programmatically register your filter
And on this step you should have information about logged in user